pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -15,6 +15,17 @@ with haskellLib;
 
 self: super: {
 
+  # The version of cabal2nix used for generating hackage-packages.nix.
+  # Should be updated from the repository's master branch with
+  # 'nix-prefetch-git https://github.com/NixOS/cabal2nix > pkgs/development/haskell-modules/cabal2nix-latest.json'.
+  cabal2nix-latest = let
+    pin = pkgs.lib.importJSON ./cabal2nix-latest.json;
+  in
+    overrideCabal self.cabal2nix (old: {
+      version = pkgs.lib.strings.sanitizeDerivationName pin.date;
+      src = pkgs.fetchgit (pkgs.lib.getAttrs [ "url" "rev" "sha256" ] pin);
+    });
```

Comment:
It is kind of a nitpick, but it seems like this might not be safe?

The `master` version of `cabal2nix` might have added new dependencies that would cause this not to compile.

Alternatively, maybe you could use `callCabal2nix` in https://github.com/NixOS/nixpkgs/pull/86699/files#diff-2ddea12de3c3f1e658e6bd5145321a86R8 to build `hackage2nix`.  It should be fine to use IFD, since we don't care about Hydra.

Reply:
You're absolutely right, I'm not sure what I was thinking of here. Definitely not just a nitpick, this was plain wrong.
The update script now uses cabal2nix itself to create a new version.